onedal/spmd/covariance/incremental_covariance.py

```diff
@@ -0,0 +1,89 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..._device_offload import support_usm_ndarray
+from ...covariance import (
+    IncrementalEmpiricalCovariance as IncrementalEmpiricalCovariance_Batch,
+)
+from ...datatypes import _convert_to_supported, to_table
+from ...utils import _check_array
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalEmpiricalCovariance(
+    BaseEstimatorSPMD, IncrementalEmpiricalCovariance_Batch
+):
+    def _reset(self):
+        self._partial_result = super(
+            IncrementalEmpiricalCovariance_Batch, self
+        )._get_backend("covariance", None, "partial_compute_result")
```

Comment:
how does this differ than the _reset that is inherited from batch IncrementalCovariance?

Reply:
Inherited reset would call spmd _get_backend from BaseEstimatorSPMD